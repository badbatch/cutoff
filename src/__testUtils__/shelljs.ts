import { jest } from '@jest/globals';

export const clearShelljsMock = (mock: jest.MockedObject<typeof import('shelljs')>) => {
  mock.echo.mockClear();
  mock.exec.mockClear();
  mock.exit.mockClear();
};

export const shelljsMock = () => ({
  default: {
    echo: jest.fn(),
    exec: jest.fn(),
    exit: jest.fn(),
  },
});
