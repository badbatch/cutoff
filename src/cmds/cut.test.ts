import shelljs from 'shelljs';

const { exec } = shelljs;

describe('cut', () => {
  it('should return the correct terminal output', () => {
    expect(exec('node --trace-warnings ./bin/cutoff.mjs cut --help', { silent: true }).stdout).toMatchSnapshot();
  });
});
