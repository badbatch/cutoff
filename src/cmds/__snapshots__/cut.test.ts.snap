// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cut should return the correct terminal output 1`] = `
"cutoff.mjs cut <type>

Cut release to current branch

Positionals:
  type  The release type: major | premajor | minor | preminor | patch | prepatch
        | prerelease                                         [string] [required]

Options:
  --version        Show version number                                 [boolean]
  --help           Show help                                           [boolean]
  --tag            The release tag: alpha | beta | unstable             [string]
  --dry-run        The release tag: alpha | beta | unstable            [boolean]
  --force          Increment version regardless of files changed       [boolean]
  --preid          The pre release ID                                   [string]
  --skip-posthook  To skip post version lifecycle hook                 [boolean]
  --skip-prehook   To skip pre version lifecycle hook                  [boolean]
"
`;
